### ISSUE TITLE
When compiled with --release Kemal crashes on empty path

### ISSUE THREAD

**Issue by bararchy** [kemalcr/kemal#399]
1. Compile Kemal project with `--release --no-debug`  
2. run the project and access `http://127.0.0.1:3000` (no trailing `/`)  
3. application crash with: 

```crystal
[production] Kemal is ready to lead at http://0.0.0.0:3000
accept: Invalid argument (Errno)
0x55caaccdbc6d: ??? at ??
0x55caacdf7687: ??? at ??
0x55caaccfd001: main at ??
0x7f1026eaef6a: __libc_start_main at ??
0x55caaccd7fca: _start at ??
0x0: ??? at ??

```

**Comment by crisward** [comment]
I've tried this through the various commits to kemal, it only starts happening after this commit 00217d9545d959fa64030c99ec71214455fb7369

**Comment by sdogruyol** [comment]
https://github.com/kemalcr/kemal/commit/00217d9545d959fa64030c99ec71214455fb7369#diff-adda3818ba95acd2e2d343f60110bc2bR11 may be the culprit (not sure)

**Comment by crisward** [comment]
Found it, I'm guessing it's a crystal bug. But I've got a work around.

It may also be worth updating the travis tests to run with and without --release.

**Comment by bararchy** [comment]
@crisward thanks for looking at it :) also , might be worth to mention upstream

**Comment by crisward** [comment]
Tried to recreate outside of kemal, no luck.

**Comment by sdogruyol** [comment]
Fixed via #407
